sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CertificateValidationCheck.cs

```diff
@@ -0,0 +1,299 @@
+
+        private static readonly DiagnosticDescriptor rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            //Handling of += delegate syntax
+            context.RegisterSyntaxNodeActionInNonGenerated(c => CheckAddHandlerSyntax(c), SyntaxKind.AddAssignmentExpression);
+
+            //Handling of constructor parameter syntax (SslStream)
+            context.RegisterSyntaxNodeActionInNonGenerated(c => CheckConstructorParameterSyntax(c), SyntaxKind.ObjectCreationExpression);
+        }
+
+        private void CheckAddHandlerSyntax(SyntaxNodeAnalysisContext c)
+        {
+            var leftIdentifier = ((AssignmentExpressionSyntax)c.Node).Left.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().LastOrDefault();
+            var right = ((AssignmentExpressionSyntax)c.Node).Right;
+            if (leftIdentifier != null && right != null
+                && c.SemanticModel.GetSymbolInfo(leftIdentifier).Symbol is IPropertySymbol left
+                && IsValidationDelegateType(left.Type))
+            {
+                TryReportLocations(new InspectionContext(c), leftIdentifier.GetLocation(), right);
+            }
+        }
+
+        private void CheckConstructorParameterSyntax(SyntaxNodeAnalysisContext c)
+        {
+            if (c.SemanticModel.GetSymbolInfo(c.Node).Symbol is IMethodSymbol ctor)
+            {
+                CSharpMethodParameterLookup methodParamLookup = null;       //Cache, there might be more of them
+                foreach (var param in ctor.Parameters.Where(x => IsValidationDelegateType(x.Type)))
+                {
+                    methodParamLookup = methodParamLookup ?? new CSharpMethodParameterLookup((c.Node as ObjectCreationExpressionSyntax).ArgumentList, ctor);
+                    if (methodParamLookup.TryGetSymbolParameter(param, out var argument))
+                    { //For Lambda expression extract location of the parentheses only to separate them from secondary location of "true"
+                        var primaryLocation = ((argument.Expression is ParenthesizedLambdaExpressionSyntax Lambda) ? (SyntaxNode)Lambda.ParameterList : argument).GetLocation();
+                        TryReportLocations(new InspectionContext(c), primaryLocation, argument.Expression);
+                    }
+                }
+            }
+        }
+
+        private void TryReportLocations(InspectionContext c, Location primaryLocation, ExpressionSyntax expression)
+        {
+            var locations = ArgumentLocations(c, expression);
+            if (!locations.IsEmpty)
+            {   //Report both, assignemnt as well as all implementation occurances
```

Comment:
nitpick: same here, to be consistent with the codebase the comment should be on next line